File: src/lessons/welcome/bat/bool2/InOrderEntity.java
package lessons.welcome.bat.bool2;

import static plm.core.ValueSerializer.deserialize;
import static plm.core.ValueSerializer.serialize;

import plm.universe.bat.BatEntity;

public class InOrderEntity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(inOrder((Integer)param[0], (Integer)param[1], (Integer)param[2], (Boolean)param[3])));
    }
  }

  /* BEGIN TEMPLATE */
  boolean inOrder(int a, int b, int c, boolean bOk)
  {
    /* BEGIN SOLUTION */
    return (bOk || (b > a)) && (c > b);
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}

File: src/lessons/welcome/bat/bool2/InOrderEntity.scala
package lessons.welcome.bat.bool2

import plm.universe.bat.BatEntity
import plm.universe.bat.BatTest

class InOrderEntity extends BatEntity {

    override def run() {
      import plm.core.ValueSerializer._

      val count = getTestCount()
      for (i <- 0 to count -1) {
        val param = deserialize(getTest(i)).asInstanceOf[Array[Object]]
      	setTestResult(i, serialize(inOrder(param(0).asInstanceOf[Int], param(1).asInstanceOf[Int], param(2).asInstanceOf[Int], param(3).asInstanceOf[Boolean]) ))
	  }
	}

	/* BEGIN TEMPLATE */
	def inOrder(a:Int, b:Int, c:Int, bOk:Boolean):Boolean = {
	/* BEGIN SOLUTION */
		return (bOk || (b > a)) && (c > b)
	/* END SOLUTION */
	}
	/* END TEMPLATE */
}
